layer/context.py

```diff
@@ -7,20 +7,45 @@
 from layer.training.base_train import BaseTrain
 
 
+# We store the active context temporarily so it can be used within
+# either a @layer decorated function or soon via `with layer.model() as context:`
+_ACTIVE_CONTEXT: Optional["Context"] = None
+
+
+def set_active_context(context: "Context") -> None:
+    global _ACTIVE_CONTEXT
+    _ACTIVE_CONTEXT = context
+
+
+def reset_active_context() -> None:
+    global _ACTIVE_CONTEXT
+    _ACTIVE_CONTEXT = None
+
+
+def get_active_context() -> Optional["Context"]:
+    """
+    Returns the active context object set from the active computation.
+    Used in local mode to identify which context to log resources to.
+
+    @return:  active context object
+    """
+    return _ACTIVE_CONTEXT
+
+
 class Context:
```

Comment:
Can we add an `__enter__` and `__exit__` to this class? With those, we wouldn't need to `set_active_context` or `reset_active_context` anywhere other than this file.

Reply:
in the description im saying i would do this in a 2nd PR :) so i limit potential conflicts in this one

for exactly this reason yes